tt_metal/CMakeLists.txt

```diff
@@ -31,6 +17,16 @@ target_link_libraries(
         magic_enum
         fmt
```

Comment:
Should we change this to `fmt::fmt-header-only` ?

Thats what this project was previously doing (but with vendored code), and tt-umd also aims to do that.

Reply:
💯 
But I'll make a dedicated PR for that, since it, too, will sweep a number of places.  I'd rather keep things more focused for easier review.  Lots more to clean up.